packages/common/src/components/popover/Popover.vue

```diff
@@ -22,10 +23,38 @@ export default defineComponent({
     const isOpen = ref(false);
     const root = ref<HTMLElement | null>(null);
 
-    const toggle = () => (isOpen.value = !isOpen.value);
-    const open = () => (isOpen.value = true);
+    const toggle = () => (
+      (isOpen.value = !isOpen.value), nextTick(() => _handlePopoverPosition())
+    );
+    const open = () => (
+      (isOpen.value = true), nextTick(() => _handlePopoverPosition())
+    );
     const close = () => (isOpen.value = false);
 
+    const _handlePopoverPosition = () => {
+      if (!isOpen.value) {
+        return;
+      }
+      const popover = document.getElementById('popover');
```

Comment:
This isn't safe because there could be more than one popover in the DOM. I think a [template ref](https://v3.vuejs.org/guide/component-template-refs.html) is what we need here instead.